smartsim/_core/mli/infrastructure/worker/worker.py

```diff
@@ -77,66 +78,33 @@ def __init__(
         self.failed = failed
 
 
-class LoadModelResult:
-    """A wrapper around a loaded model"""
+@dataclasses.dataclass
+class TensorResult:
+    """A result from a tensor operation in the pipeline"""
 
-    def __init__(self, model: t.Any) -> None:
-        """Initialize the object"""
-        self.model = model
-
-
-class TransformInputResult:
-    """A wrapper around a transformed input"""
-
-    def __init__(self, result: t.Any) -> None:
-        """Initialize the object"""
-        self.transformed = result
+    tensor: t.Any
+    shape: t.List[int]
+    order: str
+    dtype: str
 
 
-class ExecuteResult:
-    """A wrapper around inference results"""
-
-    def __init__(self, result: t.Any) -> None:
-        """Initialize the object"""
-        self.predictions = result
+_T = t.TypeVar("_T")
 
 
-class FetchInputResult:
-    """A wrapper around fetched inputs"""
+@dataclasses.dataclass
+class StageResult(t.Generic[_T]):
+    """Generic base class for results from a stage in the pipeline"""
 
-    def __init__(self, result: t.List[bytes]) -> None:
-        """Initialize the object"""
-        self.inputs = result
+    result: _T
 
 
-class TransformOutputResult:
-    """A wrapper around inference results transformed for transmission"""
-
-    def __init__(
-        self, result: t.Any, shape: t.List[int], order: str, dtype: str
-    ) -> None:
-        """Initialize the OutputTransformResult"""
-        self.outputs = result
-        self.shape = shape
-        self.order = order
-        self.dtype = dtype
-        # todo: determine if each output must have an individual (shape, order, dtype)
-
-
-class CreateInputBatchResult:
-    """A wrapper around inputs batched into a single request"""
-
-    def __init__(self, result: t.Any) -> None:
-        """Initialize the object"""
-        self.batch = result
-
-
-class FetchModelResult:
-    """A wrapper around raw fetched models"""
-
-    def __init__(self, result: bytes) -> None:
-        """Initialize the object"""
-        self.model_bytes = result
+LoadModelResult = StageResult[t.Any]
+TransformInputResult = StageResult[t.Any]
+ExecuteResult = StageResult[t.Any]
+FetchInputResult = StageResult[t.Any]
```

Comment:
Do we want to start nailing down these `t.Any` types or is that for another day?